apple/swift-docc-render
# Improve swift declaration formatting

Bug/issue #, if applicable: 88594221

## Summary

This implements a new formatting strategy for Swift declarations whereby each parameter of a multi-parameter symbol will be broken onto its own line and indented.

This should be an improvement over the way Swift declarations are currently rendered where they simply wrap at arbitrary points with some indentation. It should be easier for viewers of the documentation to visually and mentally scan/parse functions/initializers/etc for their parameters when symbols are presented in this manner.

**Example:**

Before:
![Screen Shot 2022-02-04 at 13 42 46](https://user-images.githubusercontent.com/212918/152883197-efe183e1-9c23-4de5-9f8d-91f45fb7222d.png)
After:
![Screen Shot 2022-02-04 at 13 42 57](https://user-images.githubusercontent.com/212918/152883214-e583bb9d-c266-4a19-a64a-774a36818951.png)

## Testing

Steps:
1. Download/unzip [ExamplePackage.doccarchive.zip](https://github.com/apple/swift-docc-render/files/8019198/ExamplePackage.doccarchive.zip) and configure `VUE_DEV_SERVER_PROXY=/path/to/ExamplePackage.doccarchive` (or use an alternate fixture with Swift symbols)
2. Run the dev server with `npm run serve` and use the example docs at http://localhost:8080/documentation/examplepackage to verify that symbols are formatted as described above or compare with existing data.

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.
